crowbar_framework/app/controllers/sanities_controller.rb

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright 2016, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class SanitiesController < ApplicationController
+  skip_before_filter :sanity_checks
+  before_filter :hide_navigation
+
+  def show
+    sanity_check_errors = Rails.cache.read(:sanity_check_errors)
+    @errors = if sanity_check_errors.nil? || sanity_check_errors.empty?
+      Crowbar::Sanity.check
```

Comment:
Hrm, do we need to run the checks again if it's empty? Doesn't empty mean "no error" (and not "nothing in the cache")?